# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../nxcomp/gcc34.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

This is a fix for the really ugly _hack_ to get the code compiled with recent
GCC's STLs, ...

Imported from FreeBSD via FreeNX-kNX to replace my own first quick hack fix.

  - Rene Rebe <rene@exactcode.de>

--- nxcomp/Types.h.orig	Wed Sep  1 21:18:18 2004
+++ nxcomp/Types.h	Wed Sep  1 21:21:41 2004
@@ -74,11 +74,11 @@
 
   void clear()
   {
-    #if defined(__STL_USE_STD_ALLOCATORS) || defined(__GLIBCPP_INTERNAL_VECTOR_H)
+    #if defined(__STL_USE_STD_ALLOCATORS) || defined(_VECTOR_H)
 
-    #if defined(__GLIBCPP_INTERNAL_VECTOR_H)
+    #if defined(_VECTOR_H)
 
-    _Destroy(_M_start, _M_finish);
+    _Destroy(_M_impl._M_start, _M_impl._M_finish);
 
     #else
 
@@ -86,9 +86,9 @@
 
     #endif
 
-    _M_deallocate(_M_start, _M_end_of_storage - _M_start);
+    _M_deallocate(_M_impl._M_start, _M_impl._M_end_of_storage - _M_impl._M_start);
 
-    _M_start = _M_finish = _M_end_of_storage = 0;
+    _M_impl._M_start = _M_impl._M_finish = _M_impl._M_end_of_storage = 0;
 
     #else
 
@@ -108,21 +108,21 @@
 
   void clear()
   {
-    #if defined(__STL_USE_STD_ALLOCATORS) || defined(__GLIBCPP_INTERNAL_VECTOR_H)
+    #if defined(__STL_USE_STD_ALLOCATORS) || defined(_VECTOR_H)
 
-    #if defined(__GLIBCPP_INTERNAL_VECTOR_H)
+    #if defined(_VECTOR_H)
 
-    _Destroy(_M_start, _M_finish);
+    _Destroy(_M_impl._M_start, _M_impl._M_finish);
 
     #else
 
-    destroy(_M_start, _M_finish);
+    destroy(_M_impl._M_start, _M_impl._M_finish);
 
     #endif
 
-    _M_deallocate(_M_start, _M_end_of_storage - _M_start);
+    _M_deallocate(_M_impl._M_start, _M_impl._M_end_of_storage - _M_impl._M_start);
 
-    _M_start = _M_finish = _M_end_of_storage = 0;
+    _M_impl._M_start = _M_impl._M_finish = _M_impl._M_end_of_storage = 0;
 
     #else
 
